TASK: Adjust to "serializable commands"

bwaidelich opened the pull request:
Related: https://github.com/neos/neos-development-collection/issues/5353
Related: https://github.com/neos/neos-development-collection/pull/5348

bwaidelich committed:
TASK: Adjust to "serializable commands"

Related: https://github.com/neos/neos-development-collection/issues/5353
Related: https://github.com/neos/neos-development-collection/pull/5348

mhsdesign committed:
TASK: Adjust to overhauled api of `CommandThatFailedDuringRebase`

mhsdesign committed:
TASK: Extract conflict information from event and not possibly internal command

mhsdesign committed:
TASK: Adjust to rename `EventThatFailedDuringRebase` to `ConflictingEvent`

mhsdesign commented on Classes/Infrastructure/ContentRepository/ConflictsFactory.php:
```diff
@@ -130,67 +124,42 @@ private function createConflictFromCommandThatFailedDuringRebase(
-            $command instanceof ChangeNodeAggregateType =>
+        $dimensionSpacePoint = match ($event::class) {
+            NodeAggregateWasMoved::class =>
+                $event->succeedingSiblingsForCoverage->toDimensionSpacePointSet()->getFirst(),
```
todo this is maybe cheesy? We look into the `succeedingSiblingsForCoverage` ???

bwaidelich replied:
Nasty.. I created an issue for that https://github.com/neos/neos-development-collection/issues/5360

mhsdesign commented on Classes/Infrastructure/ContentRepository/ConflictsFactory.php:
```diff
@@ -130,67 +124,42 @@ private function createConflictFromCommandThatFailedDuringRebase(
+            NodeAggregateWithNodeWasCreated::class =>
+                $event->originDimensionSpacePoint->toDimensionSpacePoint(),
+            NodeReferencesWereSet::class =>
+                $event->affectedSourceOriginDimensionSpacePoints->toDimensionSpacePointSet()->getFirst(),
```
not ideal that there is only a set here and not a `originDimensionSpacePoint` like in `SetSerializedNodeProperties`?

mhsdesign committed:
TASK: Adjust to rename `EventThatFailedDuringRebase` to `ConflictingEvent` 2

mhsdesign committed:
BUGFIX: No conflicts are shown

Regression from 36a5de71e2f1d9da88d64bb8ede316bcd47083fe as the deduplication did not work.

mhsdesign reviewed (approved):
But it seems to work so far despite the not perfect `getFirst` things ;)

mhsdesign committed:
TASK: Adjust to use `firstDimensionSpacePoint`
